tests/foreman/sys/test_katello_certs_check.py

```diff
@@ -107,24 +111,54 @@ def test_positive_validate_katello_certs_check_output(self):
             self.validate_output(result)
 
     @destructive
-    @stubbed()
     def test_positive_update_katello_certs(self):
         """Update certificates on a currently running satellite instance
 
         :id: 0ddf6954-dc83-435e-b156-b567b877c2a5
 
         :steps:
 
-            1. Generate custom certs
-            2. Run katello-certs-check with the required valid arguments
+            1. Use Jenkins provided custom certs
+            2. Assert hammer ping reports running Satellite
             3. Update satellite with custom certs
-            4. Assert output is having correct capsule-certs-generate commands
+            4. Assert output does not report SSL certificate error
+            5. Assert all services are running
+
 
         :expectedresults: Katello-certs should be updated.
 
-        :CaseAutomation: notautomated
+        :CaseAutomation: automated
         """
 
+        with get_connection(timeout=600) as connection:
+            # Check for hammer ping SSL cert error
+            result = connection.run('hammer ping')
+            assert result.return_code == 0, 'Hammer Ping fail'
+            result = connection.run(
+                'satellite-installer --scenario satellite '
+                '--certs-server-cert /tmp/server.valid.crt '
+                '--certs-server-key /tmp/server.key '
+                '--certs-server-ca-cert /tmp/rootCA.pem '
+                '--certs-update-server --certs-update-server-ca')
+            # assert no hammer ping SSL cert error
+            result = connection.run('hammer ping')
+            assert 'SSL certificate verification failed' not in result.stdout
+            assert 'ok' in result.stdout[1]
+            # assert all services are running
+            result = connection.run('foreman-maintain health check --label services-up')
+            assert 'OK' in result.stdout[2]
+        # revert to original certs
```

Comment:
@swadeley It won't revert back to original certs if above step failed and keep the satellite in broken state. Please add it in teardown or try..finally block.

Reply:
OK, try finally it is then.
Thank  you.